packages/indexer/src/services/relayer.service.ts

```diff
@@ -68,97 +91,148 @@ export class RelayerService {
-	//     await RewardPointsService.assignRewardToRelayer({
-	//      chain,
-	//      is_success: false,
-	//      earnerType: ProtocolParticipant.RELAYER,
-	//      relayer_address: relayer_id,
-	//      transaction_hash,
-	//     });
-	//    }
-
-	//    await relayer.save();
-	//    await relayer_chain_stats.save();
-
-	//    logger.info(
-	//     `Relayer: ${relayer_id} updated successfully for chain: ${chain}`
-	//    );
-	//   } catch (error) {
-	//    logger.error(
-	//     `Error while handling PostRequest/PostResponse transaction relayer updates: ${JSON.stringify(
-	//      error
-	//     )}`
-	//    );
-	//   }
-	//  }
+	/**
+	 * Computes relayer specific stats from the handlePostRequest/handlePostResponse transactions on the handlerV1 contract
+	 */
+	static async handlePostRequestOrResponseTransaction(
+		chain: string,
+		transaction: HandlePostRequestsTransaction | HandlePostResponsesTransaction,
+	): Promise<void> {
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				transaction,
+			})}`,
+		)
+		const { from: relayer_id, hash: transaction_hash, blockHash } = transaction
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				relayer_id,
+				transaction_hash,
+				blockHash,
+			})}`,
+		)
+		const receipt = await transaction.receipt()
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				receipt,
+			})}`,
+		)
+		const { status, gasUsed, effectiveGasPrice } = receipt
+
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				relayer_id,
+				chain,
+				transaction_hash,
+				status,
+				gasUsed: gasUsed.toString(),
+				effectiveGasPrice: effectiveGasPrice.toString(),
+			})}`,
+		)
+
+		const nativeCurrencyPrice = await PriceHelper.getNativeCurrencyPrice(chain)
+
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				nativeCurrencyPrice: nativeCurrencyPrice.toString(),
+			})}`,
+		)
+
+		let gasFee = BigInt(effectiveGasPrice) * BigInt(gasUsed)
+
+		logger.info(
+			`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+				gasFee: gasFee.toString(),
+			})}`,
+		)
+
+		// Add the L1 Gas Used for L2 chains
+		if (GET_ETHEREUM_L2_STATE_MACHINES().includes(chain)) {
+			if (!(receipt as any).l1Fee) {
+				logger.error(
+					`Could not find l1Fee in transaction receipt: ${JSON.stringify({
+						chain,
+						transactionHash: transaction.hash,
+					})}`,
+				)
+			}
+			const l1Fee = BigInt((receipt as any).l1Fee ?? 0)
+			gasFee += l1Fee
+			logger.info(
+				`Handling PostRequest/PostResponse Transaction Relayer Update: ${JSON.stringify({
+					l1Fee: l1Fee.toString(),
+				})}`,
+			)
+		}
+
+		const gasFeeInEth = Number(gasFee) / Number(BigInt(10 ** 18))
+		const usdFee = (gasFee * nativeCurrencyPrice) / BigInt(10 ** 18)
```

Comment:
Also is nativeCurrencyPrice in 18 decimals or just a regular float?
```suggestion
		const usdFee = (gasFeeInEth * nativeCurrencyPrice) / BigInt(10 ** 18)
```